src/libmongoc/tests/unified/entity-map.c

```diff
@@ -269,71 +271,125 @@ command_succeeded (const mongoc_apm_command_succeeded_t *succeeded)
+      find (
+         key ("ignoreCommandMonitoringEvents"),
+         if (not(type (array)),
+             then (error ("'ignoreCommandMonitoringEvents' must be an array"))),
+         visitEach (if (not(type (utf8)),
+                        then (error ("Every 'ignoreCommandMonitoringEvents' "
+                                     "element must be a string")))),
+         storeDocDupPtr (entity->ignore_command_monitoring_events)),
+      // Parse the serverApi, if present
+      find (
+         key ("serverApi"),
+         if (not(type (doc)), then (error ("'serverApi' must be a document"))),
+         parse ( // The "version" string is required first:
+            find (keyWithType ("version", utf8), do({
+                     mongoc_server_api_version_t ver;
+                     if (!mongoc_server_api_version_from_string (bsonAs (cstr),
+                                                                 &ver)) {
+                        bsonParseError = "Invalid serverApi.version string";
+                     } else {
+                        api = mongoc_server_api_new (ver);
+                     }
+                  })),
+            else(error ("Missing 'version' property in 'serverApi' object")),
+            // Toggle strictness:
+            find (key ("strict"),
+                  if (not(type (bool)),
+                      then (error ("'serverApi.strict' must be a bool"))),
+                  do(mongoc_server_api_strict (api, bsonAs (bool)))),
+            // Toggle deprecation errors:
+            find (
+               key ("deprecationErrors"),
+               if (not(type (bool)),
+                   then (error ("serverApi.deprecationErrors must be a bool"))),
+               do(mongoc_server_api_deprecation_errors (api, bsonAs (bool)))))),
+      // Toggle observation of sensitive commands
+      find (key ("observeSensitiveCommands"),
+            if (not(type (bool)),
+                then (error ("'observeSensitiveCommands' must be a bool"))),
+            do({
+               bool *p = entity->observe_sensitive_commands =
+                  BSON_ALIGNED_ALLOC0 (bool);
+               *p = bsonAs (bool);
+            })),
+      // Which events should be available as entities:
+      find (key ("storeEventsAsEntities"),
+            if (not(type (array)),
+                then (error ("'storeEventsAsEntities' must be an array"))),
+            visitEach (if (not(type (utf8)),
+                           then (error ("Every 'storeEventsAsEntities' element "
+                                        "must be a string")))),
+            /* TODO: CDRIVER-3867 Comprehensive Atlas Testing */
+            error ("'storeEventsAsEntities' is not yet supported")),
+      visitOthers (dupPath (errpath),
+                   errorf (err,
+                           "At [%s]: Unknown key '%s' given in entity options",
+                           errpath,
+                           bson_iter_key (&bsonVisitIter))));
```

Comment:
```suggestion
      visitOthers (dupPath (errpath),
                   errorf (err,
                           "At [%s]: Unknown key '%s' given in entity options",
                           errpath,
                           bson_iter_key (&bsonVisitIter)),
                   do(bson_free (errpath))));
```
This suggestion is also a question: given `dupPath` allocates and responsibility for freeing is on caller, is the addition of this `do(bson_free (errpath))` appropriate? (Besides the fact that this test unconditionally aborts if this path is taken.)

Reply:
It is safe to add the free(). I only omitted it because the program definitely terminates on this condition.